plugins/modules/rulebook_activation.py

```diff
@@ -555,22 +601,39 @@ def main() -> None:
         except EDAError as e:
             module.fail_json(msg=f"Failed to copy rulebook activation: {e}")
 
+    # Parse credential IDs in existing activation
     if activation:
-        module.exit_json(
-            msg=f"A rulebook activation with name: {name} already exists. "
-            "The module does not support modifying a rulebook activation.",
-            changed=False,
-            id=activation["id"],
-        )
+        credential_ids = [
+            credential_id["id"] for credential_id in activation["eda_credentials"]
+        ]
+        activation["eda_credentials"] = credential_ids
 
     # Activation Data that will be sent for create/update
     activation_params = create_params(module, controller, is_aap_24=is_aap_24)
-    activation_params["name"] = name
+    activation_params["name"] = (
+        new_name
+        if new_name
+        else (controller.get_item_name(activation) if activation else name)
+    )
+
+    if state in ("enabled", "disabled") and activation:
+        activation_params["is_enabled"] = True if state == "enabled" else False
+
+        # Handle disable separately, as enable is treated by the backend.
+        try:
+            if activation["is_enabled"] and not activation_params["is_enabled"]:
+                controller.post_endpoint(
+                    endpoint=f"activations/{activation['id']}/disable"
+                )
+                module.exit_json(changed=True)
```

Comment:
After the activation is disabled, should you always exit the module or continue if there are other attributes to be updated in `activation_params`?

Reply:
@bzwei This is actually a design decision. Whether we:
  - treat them as separate things and perform enable/disable, and edit, in different tasks
  - allow an activation to be enabled/disabled/edited in one shot, which honors the state management aspect of the module that we want to have

cc: @Alex-Izquierdo @alinabuzachis I believe we talked about this, but given we are going for stateful management of activations, we should stick with option no. 2, right?

Reply:
We don't want to be operational anymore. The task declares a state and the module should ensure the activation is the declared state unless the declaration is invalid.  The point here is that the activation requires to be disabled to be edited. So once the state is disabled is should be updated according to what it is declared in the task. 

If you try to edit an activation that it is in enabled state the backend would return and error and the module should fail with that error.

Reply:
@Alex-Izquierdo So if one tries to disable and edit the activation at the same time, it's a valid state, right? The module will have to disable it and then send the update.

What if the activation takes a bit longer to be disabled and the module tries to perform the update but when it does, the activation is still not disabled? We would need to wait for the disabling to go through, right? Otherwise, the backend would return an error

Reply:
To allow this combination, we need the wait_for implementation, either a default timeout or user configurable. If wait_for is a separate feature, this PR should not support disable and editing combination.

Reply:
Yeah, is the same that I said, just I'm stupid and did it in the wrong thread. :(
https://github.com/ansible/event-driven-ansible/pull/409#discussion_r2010392991